include/crocoddyl/multibody/actions/constraint-fwddyn.hxx

```diff
@@ -0,0 +1,196 @@
+  }
+  if (static_cast<std::size_t>(u.size()) != nu_) {
+    throw_pretty("Invalid argument: "
+                 << "u has wrong dimension (it should be " + std::to_string(nu_) + ")");
+  }
+
+  DifferentialActionDataConstraintFwdDynamicsTpl<Scalar>* d =
+      static_cast<DifferentialActionDataConstraintFwdDynamicsTpl<Scalar>*>(data.get());
+  const Eigen::VectorBlock<const Eigen::Ref<const VectorXs>, Eigen::Dynamic> q = x.head(state_->get_nq());
+  const Eigen::VectorBlock<const Eigen::Ref<const VectorXs>, Eigen::Dynamic> v = x.tail(state_->get_nv());
+
+  // Computing the forward dynamics with the holonomic constraints defined by the contact model
+
+  actuation_->calc(d->multibody.actuation, x, u);
+  // TODO: Add mu
+  d->xout = pinocchio::constraintDynamics(pinocchio_, d->pinocchio, q, v, d->multibody.actuation->tau, contacts_,
+                                          d->multibody.contacts);  //, mu_contacts);
+  pinocchio::jacobianCenterOfMass(pinocchio_, d->pinocchio, false);
+  // Computing the cost value and residuals
+  costs_->calc(d->costs, x, u);
+  d->cost = d->costs->cost;
+}
+
+template <typename Scalar>
+void DifferentialActionModelConstraintFwdDynamicsTpl<Scalar>::calcDiff(
+    const boost::shared_ptr<DifferentialActionDataAbstract>& data, const Eigen::Ref<const VectorXs>& x,
+    const Eigen::Ref<const VectorXs>& u) {
+  if (static_cast<std::size_t>(x.size()) != state_->get_nx()) {
+    throw_pretty("Invalid argument: "
+                 << "x has wrong dimension (it should be " + std::to_string(state_->get_nx()) + ")");
+  }
+  if (static_cast<std::size_t>(u.size()) != nu_) {
+    throw_pretty("Invalid argument: "
+                 << "u has wrong dimension (it should be " + std::to_string(nu_) + ")");
+  }
+
+  const std::size_t& nv = state_->get_nv();
+  const Eigen::VectorBlock<const Eigen::Ref<const VectorXs>, Eigen::Dynamic> q = x.head(state_->get_nq());
+  const Eigen::VectorBlock<const Eigen::Ref<const VectorXs>, Eigen::Dynamic> v = x.tail(nv);
+
+  DifferentialActionDataConstraintFwdDynamicsTpl<Scalar>* d =
+      static_cast<DifferentialActionDataConstraintFwdDynamicsTpl<Scalar>*>(data.get());
+
+  // Computing the dynamics derivatives
+  actuation_->calcDiff(d->multibody.actuation, x, u);
+  pinocchio::computeConstraintDynamicsDerivatives(
+      pinocchio_, d->pinocchio, contacts_, d->multibody.contacts, d->Fx.leftCols(nv), d->Fx.rightCols(nv),
+      d->pinocchio.ddq_dtau, d->pinocchio.dlambda_dq, d->pinocchio.dlambda_dv, d->pinocchio.dlambda_dtau);
+
+  d->Fu.noalias() = d->pinocchio.ddq_dtau * d->multibody.actuation->dtau_du;
+  costs_->calcDiff(d->costs, x, u);
+}
+
+template <typename Scalar>
+boost::shared_ptr<DifferentialActionDataAbstractTpl<Scalar> >
+DifferentialActionModelConstraintFwdDynamicsTpl<Scalar>::createData() {
+  boost::shared_ptr<DifferentialActionDataAbstractTpl<Scalar> > data =
+      boost::make_shared<DifferentialActionDataConstraintFwdDynamics>(this);
```

Comment:
This won't compile with vectorization support. For that, we need to use `Eigen::aligned_allocator` as suggested below
```suggestions
  return boost::allocate_shared<Data>(Eigen::aligned_allocator<Data>(), this);
```

This code assumes that you have defined the `Data` type in the `hpp` file, which I highly suggest to do it as it is follows the Crocoddyl code style.